### PR TITLE
Restructured Chart

### PR DESCRIPTION
**Pull request by nshankar13**
Separated this chart into two new ones: osm-arc and osm. OSM is a dependency of osm-arc, and osm-arc values will override those in the values.yaml of osm. Osm-arc will have templates, such as osm-label, that are specific for arc environments. 

Now the workflow for installing the chart via helm will look something like this: 

`helm dependency update osm-arc`
`helm install osm osm-arc --namespace arc-osm-system`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,7 +23,8 @@ e2e-helm-deploy:
 	mkdir -p .staging/helm
 	curl https://get.helm.sh/helm-v${HELM_VERSION}-linux-amd64.tar.gz > .staging/helm/helmbin.tar.gz
 	cd .staging/helm && tar -xvf helmbin.tar.gz
-	./.staging/helm/linux-amd64/helm install osm ./charts/osm --namespace arc-osm-system
+	./.staging/helm/linux-amd64/helm dependency update ./charts/osm-arc
+	./.staging/helm/linux-amd64/helm install osm ./charts/osm-arc --namespace arc-osm-system
```
Comment (ritazh) [inline Makefile:27]:
```suggestion
	./.staging/helm/linux-amd64/helm install osm ./charts/osm-arc --namespace arc-osm-system --dependency-update
```

File: Makefile [hunk 2]
```diff
@@ -23,7 +23,8 @@ e2e-helm-deploy:
 	mkdir -p .staging/helm
 	curl https://get.helm.sh/helm-v${HELM_VERSION}-linux-amd64.tar.gz > .staging/helm/helmbin.tar.gz
 	cd .staging/helm && tar -xvf helmbin.tar.gz
-	./.staging/helm/linux-amd64/helm install osm ./charts/osm --namespace arc-osm-system
+	./.staging/helm/linux-amd64/helm dependency update ./charts/osm-arc
```
Comment (ritazh) [inline Makefile:26]: you can remove this

File: Makefile [hunk 3]
```diff
@@ -23,7 +23,8 @@ e2e-helm-deploy:
...
+	./.staging/helm/linux-amd64/helm install osm ./charts/osm-arc --namespace arc-osm-system
 
 test-e2e:
```
Comment (ritazh) [inline Makefile:29]: We should add a new Makefile target similar to this [example](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/Makefile#L130-L139). We can run this target before cutting a release of osm-arc to ensure the osm chart has been updated and included as part of the new release of osm-arc chart.

File: charts/osm-arc/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+apiVersion: v2
+name: osm-arc
+description: A Helm chart for Kubernetes
```
Comment (ritazh) [inline charts/osm-arc/Chart.yaml:3]:
```suggestion
description: A Helm chart to install the OSM control plane on Azure Arc enabled Kubernetes clusters.
```

File: charts/osm-arc/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+appVersion: v0.5.0
+
+dependencies: 
+- name: osm
+  version: "0.5.0-rc.1"
+  repository: "file://../../charts/osm"
```
Comment (ritazh) [inline charts/osm-arc/Chart.yaml:28]: Should this be https://github.com/openservicemesh/osm/tree/main/charts/osm?

File: charts/osm-arc/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.5.0
```
Comment (ritazh) [inline charts/osm-arc/Chart.yaml:18]: The value here should match the osm-arc tgz file generated.

File: charts/osm/Chart.yaml [hunk 7]
```diff
@@ -14,8 +14,8 @@ type: application
```
Comment (ritazh) [inline charts/osm/Chart.yaml:14]: I think you can add all the charts/osm files to gitignore

File: charts/osm-arc/Chart.lock [hunk 8]
```diff
@@ -0,0 +1,6 @@
+dependencies:
+- name: osm
+  repository: file://../../charts/osm
```
Comment (ritazh) [inline charts/osm-arc/Chart.lock:3]: Should this be https://github.com/openservicemesh/osm/tree/main/charts/osm?

File: charts/osm-arc/values.yaml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    ignoreNamespaces: "kube-system azure-arc arc-osm-system"
+
+osm:
+  OpenServiceMesh:
+    enableDebugServer: true
+    enablePermissiveTrafficPolicy: true
+    enableFluentbit: true
+    fluentBitImage:
+      name: fluentbit-logger
+      registry: "mcr.microsoft.com/oss/fluent"
```
Comment (ritazh) [inline charts/osm-arc/values.yaml:14]: You also need to update the [envoy image](https://github.com/openservicemesh/osm/blob/main/charts/osm/values.yaml#L11) and the [osm-controller image](https://github.com/openservicemesh/osm/blob/main/charts/osm/values.yaml#L6-L7).

File: charts/values-override.yaml [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# Default values for osm.
```
Comment (ritazh) [inline charts/values-override.yaml:1]: This file should be values.yaml.

File: charts/osm-arc/Chart.yaml (hunk 5 above)
Comment (ritazh) [reply charts/osm-arc/Chart.yaml:28]: @michelleN I noticed we dont have index.yaml in https://github.com/openservicemesh/osm/tree/main/charts/osm is that going to be an issue finding older versions of the osm chart?

**Review by michelleN — CHANGES_REQUESTED** [review]
Could you split this into separate PRs? The first PR being the one that creates the `osm-arc` chart with no dependency so we can review that separately.

**Review by michelleN — APPROVED** [review]

File: .gitignore [hunk 11]
```diff
@@ -353,3 +353,6 @@ MigrationBackup/
 
 # Ionide (cross platform F# VS Code tools) working folder
 .ionide/
+
+# Chart-tgzs generated by dependency update
+charts/osm/charts
```
Comment (ritazh) [inline .gitignore:358]: this should be `charts/osm-arc/charts` as you want to ignore the dependency chart in the new osm-arc chart. please also `git rm charts/osm-arc/charts` as you currently have those files checked in.

File: test/bats/test.bats [hunk 12]
```diff
@@ -16,7 +16,7 @@ SLEEP_TIME=1
 }
 
 @test "mutating webhook has been deployed" {
-    run wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl get mutatingwebhookconfigurations.admissionregistration.k8s.io arc-osm-webhook-osm"
+    run wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl get mutatingwebhookconfigurations.admissionregistration.k8s.io osm-webhook-osm"
```
Comment (ritazh) [inline test/bats/test.bats:19]: please open an issue in this repo to update this after the dependent chart has been updated to 0.5.1

File: charts/osm-arc/values.yaml (hunk 9 above)
Comment (ritazh) [reply charts/osm-arc/values.yaml:16]: Still missing the envoy image

File: charts/osm-arc/values.yaml [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+      tag: "v1.6.2"
+      pullPolicy: IfNotPresent
+    enforceSingleMesh: true
+    enablePrometheus: false
+    deployJaeger: false
+    customLogAnalyticsWorkspace: false
+    customerID: 49a4d43a-81ef-4607-83f6-a13872c6ae60
```
Comment (ritazh) [inline charts/osm-arc/values.yaml:23]: You should include these new fields in this values.yaml _after_ the osm dependency chart has been upgraded to 0.5.1 as they dont exist in 0.5.0 right?

File: Makefile (hunk 3 above)
Comment (ritazh) [reply Makefile:31]: created https://github.com/Azure/osm-azure/issues/21

**Review by ritazh — APPROVED** [review]
lgtm

**Comment by ritazh** [comment]
cc @snehachhabria @SanyaKochhar PTAL

**Review by snehachhabria — APPROVED** [review]

**Review by SanyaKochhar — APPROVED** [review]

**Pull request merged**